### PR TITLE
Updated/imporved Flare Module

### PR DESCRIPTION
**Pull request by Soultechnology**
Changed spawned projectile to a SOG one: 
	- audible 'POP' on illumination
	- goes inert in a puff of smoke on impact with objects/terrian
	- new launch effects
	- All colors visable as white in broad daylight
Can now fire flares from FFV seats
Can now use more items as flare ammo (like the ace throwable signal grenades)
BUG: Yellow just fires a white flare, due to the sog projectiles behaviour i think we'll need a new projectile for new colors.

[functions > PARAFL > fn_flareLight.sqf was just commented out entirely as its nolonger called or used(and i cant figure out how to push a file deletion for cleanup)]

### CODE REVIEW

File: modules/paraflare/preInitClient.sqf [hunk 1]
```diff
@@ -1,9 +1,7 @@
...
-["Paraflares", "Paraflares, old style.", "TinfoilHate &amp; StatusRed", _version] call EFUNC(FW,RegisterModule);
-
-GVAR(flareAngle) = 90;
-
-["paraflare_evt_flareLight", FUNC(flareLight)] call CBA_fnc_addEventHandler;
+#include "script_component.hpp"
+
+private _version = 0.1;
+
+["Paraflares", "Paraflares, old style.", "TinfoilHate &amp; StatusRed", _version] call EFUNC(FW,RegisterModule);
```
Comment (EM-Creations) [inline modules/paraflare/preInitClient.sqf:5]: All your name here pal.

File: modules/paraflare/functions/PARAFL/fn_flareFire.sqf [hunk 2]
```diff
@@ -1,52 +1,48 @@
...
+  [player, "tin_flareFire"] remoteExec ["say3D"];
+
+  [{
+	params ["_ammoType", "_pos", "_dir"];
+
+    private _flare = _ammoType createVehicle _pos;
+
+	_flare setVectorDirAndUp [[sin _dir * cos  GVAR(flareAngle), cos _dir * cos  GVAR(flareAngle), sin  GVAR(flareAngle)], [0,0,1]];
+
+    [_flare, "tin_flareShot"] remoteExec ["say3D"];
```
Comment (EM-Creations) [inline modules/paraflare/functions/PARAFL/fn_flareFire.sqf:45]: Where does this sound come from, is it already in the framework?

**Review by EM-Creations — CHANGES_REQUESTED** [review]
As per comments.

File: modules/paraflare/functions/PARAFL/fn_flareFire.sqf (hunk 2 above)
Comment (Soultechnology) [reply modules/paraflare/functions/PARAFL/fn_flareFire.sqf:45]: Alreayd in the framework/Paraflare module. i did not add or remove anything to do with sounds (new sounds mentioned in changes are just apart of the new projectile from SOG)

File: modules/paraflare/functions/PARAFL/fn_flareFire.sqf [hunk 3]
```diff
@@ -1,52 +1,48 @@
...
-      }] call CBA_fnc_waitUntilAndExecute;
-    }, [_flare, _magColor], 2.6] call CBA_fnc_waitAndExecute;
-  }, [_ammoType, _pos, _dir, _magColor], 0.95] call CBA_fnc_waitAndExecute;
-}, [_ammoType, _pos, _dir, _magColor], 1] call CBA_fnc_waitAndExecute;
+#include "script_component.hpp"
+
+params["_magType"];
+
+private _pos = player modelToWorld [0, 1, 0];
+_pos = [_pos select 0, _pos select 1, (_pos select 2)+1.5];
```
Comment (Soultechnology) [inline modules/paraflare/functions/PARAFL/fn_flareFire.sqf:6]: Forgot to mention in change notes. This line was changed to fix the bug with v0.1 ParaFlare where flares would fir eform gorund level alyways, preventing effective use from ontop of buildings and objects.

File: modules/paraflare/preInitClient.sqf (hunk 1 above)
Comment (EM-Creations) [reply modules/paraflare/preInitClient.sqf]: If I click through to your branch I see the update, but still don't see the update in this PR? :S

**Review by EM-Creations — APPROVED** [review]

**Pull request merged**